File: C#高级编程(第7版)源代码/502259 Code CH53/FSharp/DelegateSample.fs
namespace Wrox.ProCSharp.Languages

type DemoDelegate = delegate of int -> unit

type Demo() as this =
    static member Foo(x : int) = 
        printfn "Foo %d" x
    
    member this.Bar(x : int) =
        printfn "Bar %d" x

File: C#高级编程(第7版)源代码/502259 Code CH53/CSharp/DelegateSample.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CSharp
{
    public delegate void DemoDelegate(int x);

    class Demo
    {
        public static void Foo(int x)
        {
            Console.WriteLine("Foo {0}", x);
        }

        public void Bar(int x)
        {
            Console.WriteLine("Bar {0}", x);
        }
    }
}
